packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -69,7 +67,7 @@ class GenSnapshot {
     // iOS has a separate gen_snapshot for armv7 and arm64 in the same,
     // directory. So we need to select the right one.
     if (snapshotType.platform == TargetPlatform.ios) {
-      snapshotterPath += '_' + getNameForDarwinArch(darwinArch);
+      snapshotterPath += '_' + getNameForDarwinArch(darwinArch!);
     }
 
     return _processUtils.stream(
```

Comment:
Line 77 is the null returned from `mapFunction` that prompted that signature change.